### ISSUE TITLE
Artifactory PRO with aws-s3-v3

### ISSUE THREAD

**Issue by Andreiaotto** [jfrog/charts#1301]
**Is this a request for help?**:
Yes

---

**Which chart**:
stable/artifactory verison 7.5.7 and 7.11.5

**What happened**:
I have deployed jfrog artifactory PRO, version 7.5.7 and 7.11.5 using helm:
https://github.com/jfrog/charts/tree/master/stable/artifactory
We have aws-s3-v3 for file storage and it works fine for local repos.

If I create a remote repo and enable the Storage artifacts locally option, I get a 500 with *Could not process download request: Illegal endpoint: 'https://s3.eu-west-1.amazonaws.com'. This endpoint is not supported"*
If I disable store artifacts locally (caching option), it works fine.

Anyone had this issue?

**What you expected to happen**:
Remote repo working with caching enabled.

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

Not sure if this is the right place to ask it, but in case someone had this issue, please share.

**Comment by gitta-jfrog** [comment]
In order to use s3 with "explicit credentials", the parameter "useInstanceCredentials" (default: true) need to be set as false.
Please try to install the chart again with:

`--set artifactory.persistence.type=aws-s3-v3 \
--set artifactory.persistence.awsS3V3.region=${AWS_REGION} \
--set artifactory.persistence.awsS3V3.bucketName=${AWS_S3_BUCKET_NAME} \
--set artifactory.persistence.awsS3V3.identity=${AWS_ACCESS_KEY_ID} \
--set artifactory.persistence.awsS3V3.credential=${AWS_SECRET_ACCESS_KEY} \
--set artifactory.persistence.awsS3V3.useInstanceCredentials=false \`

You might verify the change take effect, exec into the Artifactory container, and check the binarystore.xml file /opt/jfrog/artifactory/var/etc/artifactory/binarystore.xml:
`    <provider type="s3-storage-v3" id="s3-storage-v3">
        <bucketName>XXXX</bucketName>
        <path>XXXX</path>
        <endpoint>s3.eu-west-2.amazonaws.com</endpoint>
        <testConnection>false</testConnection>
        <identity>XXX</identity>
       <credential>XXX</credential>
        <signatureExpirySeconds>300</signatureExpirySeconds>
        <region>eu-west-2</region>
        <usePresigning>false</usePresigning>
        **<useInstanceCredentials>false</useInstanceCredentials>**
        <maxConnections>50</maxConnections>`

**Comment by Andreiaotto** [comment]
Hello @gitta-jfrog 
I updated it and still the same:
```
 [ERROR] [4293f661ef7a3479] [.s.b.p.RetryBinaryProvider:146] [ttp-nio-8081-exec-10] - Failed to fetch blob 'ba0d47e466424162f11c5f7ba4ff1a89db1bb870' from next binary provider
java.lang.IllegalStateException: Illegal endpoint: 'https://s3.eu-west-1.amazonaws.com'. This endpoint is not supported
```
```
 [jfrt ] [ERROR] [4293f661ef7a3479] [currentReadCacheMissHandler:78] [ttp-nio-8081-exec-10] - Exception was thrown while getting Concurrent stream from cache for checksum ba0d47e466424162f11c5f7ba4ff1a89db1bb870
com.google.common.util.concurrent.UncheckedExecutionException: java.lang.IllegalStateException: Illegal endpoint: 'https://s3.eu-west-1.amazonaws.com'. This endpoint is not supported
```
then last - with 500
`[jfrt ] [WARN ] [4293f661ef7a3479] [.r.ArtifactoryResponseBase:125] [ttp-nio-8081-exec-10] - Sending HTTP error code 500: Could not process download request: Illegal endpoint: 'https://s3.eu-west-1.amazonaws.com'. This endpoint is not supported`

binarystore:
/opt/jfrog/artifactory/var/etc/artifactory/binarystore.xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<config version="2">
    <chain>
        <provider type="cache-fs" id="cache-fs">
            <provider type="eventual" id="eventual">
                <provider type="retry" id="retry">
                    <provider type="s3-storage-v3" id="s3-storage-v3"/>
                </provider>
            </provider>
        </provider>
    </chain>
    <provider type="cache-fs" id="cache-fs">
        <maxCacheSize>5e+09</maxCacheSize>
        <cacheProviderDir>cache</cacheProviderDir>
    </provider>
    <provider type="s3-storage-v3" id="s3-storage-v3">
        <bucketName>bucketName</bucketName>
        <path>artifactory/filestore</path>
        <endpoint>https://s3.eu-west-1.amazonaws.com</endpoint>
        <credential>Cred</credential>
        <testConnection>false</testConnection>
        <identity>Identify</identity>
        <signatureExpirySeconds>300</signatureExpirySeconds>
        <region>eu-west-1</region>
        <usePresigning>false</usePresigning>
        <useInstanceCredentials>false</useInstanceCredentials>
    </provider>
</config>

**Comment by gitta-jfrog** [comment]
I checked again, right now **Pro license** support only "s3.amazonaws.com" as AWS endpoint.
Please try to install it again and verify the endpoint in the binarystore.xml:  `<endpoint>s3.amazonaws.com</endpoint>`

**Comment by Andreiaotto** [comment]
Hi @gitta-jfrog , I tested as you suggested but still the same:

Right, for aws S3 endpoint in eu-west-1 works fine, thank you! I had the eu-west-1 in the endpoint and it was causing the issue.
Actually I deployed it in eu-west-1 to isolate the problem, our current problem in in China, so endpoint must be s3.cn-north-1.amazonaws.com.cn, according to https://docs.amazonaws.cn/en_us/aws/latest/userguide/endpoints-Beijing.html.

Without the region, the endpoint is not recognized, I get 
```
[jfrt ] [ERROR] [984079334d793c10] [j.s.S3JetS3tBinaryProvider:374] [art-init            ] - Failed to put blob '86a3d21e58480a8325974fad4cce2de04739c15f' to s3 caused by -1:null
org.jets3t.service.S3ServiceException: Request Error: artifactory-stg-cn-north-1.s3.amazonaws.com.cn: Name or service not known
```
And with the region in the s3 endpoint, we get 500, so it leads me to conclude the issue is with having the region inside the endpoint. 

Should it be working with China AWS S3?

Thank you in advance,
Andreia
